app/plugin_hooks_test.go

```diff
@@ -1131,3 +1131,42 @@ func TestHookMetrics(t *testing.T) {
 		metricsMock.AssertExpectations(t)
 	})
 }
+
+func TestOnPluginStatusesChanged(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	tearDown, pluginIds, _ := SetAppEnvironmentWithPlugins(t,
+		[]string{
+			`
+		package main
+		import (
+			"github.com/mattermost/mattermost-server/v5/plugin"
+		)
+		type MyPlugin struct {
+			plugin.MattermostPlugin
+		}
+		func (p *MyPlugin) OnPluginStatusesChanged(c *plugin.Context) {
+			p.API.GetPluginStatuses()
```

Comment:
Can we find a way to assert that this method actually gets called? The assertions below verify that `th.App.GetPluginStatuses` returns new plugin statuses, but not that a plugin was notified about same.

Take a look at `TestHookMessageHasBeenUpdated` and how it uses `LogDebug` to assert that certain events occur "inside" the plugin.